core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -403,7 +406,20 @@ private[spark] class BlockManager(
    */
   def getLocal(blockId: BlockId): Option[BlockResult] = {
     logDebug(s"Getting local block $blockId")
-    doGetLocal(blockId, asBlockResult = true).asInstanceOf[Option[BlockResult]]
+    val res = doGetLocal(blockId, asBlockResult = true).asInstanceOf[Option[BlockResult]]
```

Comment:
A shortened version of this should be a class comment. Describe what pinning means and what are the API semantics.

"we should add a pin-counting mechanism to track which blocks/pages are being read in order to prevent them from being evicted prematurely. I propose to do this in two phases: first, add a safe, conservative approach in which all BlockManager.get*() calls implicitly increment the pin count of blocks and where tasks' pins are automatically freed upon task completion (this PR)"